test/tbb/test_arena_constraints.cpp

```diff
@@ -208,3 +212,10 @@ TEST_CASE("Testing constraints_threads_per_core() reserved entry point") {
     tbb::task_arena::constraints c{};
     tbb::detail::r1::constraints_threads_per_core(c);
 }
+
+//! Using custom assertion handler to test failure on invalid constraints
+//! \brief \ref interface \ref error_guessing
+TEST_CASE("Using custom assertion handler to test failure on invalid constraints") {
+    TRY_BAD_EXPR(tbb::info::default_concurrency(tbb::task_arena::constraints{}.set_max_threads_per_core(0)),
+        "Wrong max_threads_per_core constraints field value.");
```

Comment:
I would rename `TRY_BAD_EXPR` to something reflecting the testing of custom assertion handler. Something like `TEST_CUSTOM_ASSERTION_HANDLER_WITH_BAD_EXPR` or simply `TEST_CUSTOM_ASSERTION_HANDLER`